src/sync/clients.ts

```diff
@@ -92,3 +95,55 @@ export async function setClient(
   clients.set(id, client);
   return setClients(clients, dagWrite);
 }
+
+export async function initClient(
+  dagStore: dag.Store,
+): Promise<[ClientID, Client]> {
+  return dagStore.withWrite(async dagWrite => {
+    const clients = await getClients(dagWrite);
+    const newClientID = makeUuid();
+    const bootstrapClient = [...clients.values()].reduce((prev, curr) =>
+      !prev || prev.heartbeatTimestampMs < curr.heartbeatTimestampMs
+        ? curr
+        : prev,
+    );
+
+    let newClientCommit;
+    if (bootstrapClient) {
+      const bootstrapCommit = await Commit.baseSnapshot(
+        bootstrapClient.headHash,
+        dagWrite,
+      );
+      // Copy the snapshot with one change: set last mutation id to 0.
+      newClientCommit = newSnapshot(
+        dagWrite.createChunk,
+        bootstrapCommit.meta.basisHash,
+        0 /* lastMutationID */,
+        bootstrapCommit.meta.cookieJSON,
+        bootstrapCommit.valueHash,
+        bootstrapCommit.indexes,
+      );
+    } else {
+      // No existing snapshot to bootstrap from. Create empty snapshot.
```

Comment:
This has the same effect as`initDB` in `db/write.ts`, but doesn't set the dag head `main` to the commits `hash`.  `initDB` uses the more general and complicated `db.Write#commitWithChangedKeys` code path.

I believe in Simplified Dueling Dags `db.Write` will only be used on the memdag, and not on the permdag.

Reply:
I think you are right. The perdag does use db.Commit but only in the sense that it uses it to create chunks